Sources/SwiftInspectorVisitors/Tests/TypeDescriptionSpec.swift

```diff
@@ -0,0 +1,346 @@
+
+        it("to decode the encoded type description") {
+          expect(try decoder.decode(TypeDescription.self, from: data)) == sut
+        }
+      }
+
+      context("when decoding a member type") {
+        beforeEach {
+          sut = .member(name: "Bar", baseType: .simple(name: "Foo"))
+          data = try? encoder.encode(sut)
+        }
+
+        it("to decode the encoded type description") {
+          expect(try decoder.decode(TypeDescription.self, from: data)) == sut
+        }
+      }
+
+      context("when decoding an optional type") {
+        beforeEach {
+          sut = .optional(.simple(name: "Foo"))
+          data = try? encoder.encode(sut)
+        }
+
+        it("to decode the encoded type description") {
+          expect(try decoder.decode(TypeDescription.self, from: data)) == sut
+        }
+      }
+
+      context("when decoding an implicitlyUnwrapped type") {
+        beforeEach {
+          sut = .implicitlyUnwrapped(.simple(name: "Foo"))
+          data = try? encoder.encode(sut)
+        }
+
+        it("to decode the encoded type description") {
+          expect(try decoder.decode(TypeDescription.self, from: data)) == sut
+        }
+      }
+
+      context("when decoding an array type") {
+        beforeEach {
+          sut = .array(.simple(name: "Foo"))
+          data = try? encoder.encode(sut)
+        }
+
+        it("to decode the encoded type description") {
+          expect(try decoder.decode(TypeDescription.self, from: data)) == sut
+        }
+      }
+
+      context("when decoding a dictionary type") {
+        beforeEach {
+          sut = .dictionary(key: .simple(name: "Foo"), value: .simple(name: "Bar"))
+          data = try? encoder.encode(sut)
+        }
+
+        it("to decode the encoded type description") {
+          expect(try decoder.decode(TypeDescription.self, from: data)) == sut
+        }
+      }
+
+      context("when decoding a tuple type") {
+        beforeEach {
+          sut = .tuple([.simple(name: "Foo"), .optional(.simple(name: "Bar"))])
+          data = try? encoder.encode(sut)
+        }
+
+        it("to decode the encoded type description") {
+          expect(try decoder.decode(TypeDescription.self, from: data)) == sut
+        }
+      }
+
+      context("when decoding an unknown type") {
+        beforeEach {
+          sut = .unknown(text: "Foo")
+          data = try? encoder.encode(sut)
+        }
+
+        it("to decode the encoded type description") {
+          expect(try decoder.decode(TypeDescription.self, from: data)) == sut
+        }
+      }
+
+      context("when decoding an unknown case") {
+        beforeEach {
+          data = try? encoder.encode("garbage")
+        }
+
+        it("throws") {
+          expect(try decoder.decode(TypeDescription.self, from: data)).to(throwError())
+        }
+      }
+    }
+
+    describe("typeDescription") {
+      context("when called on a TypeSyntax node representing a SimpleTypeIdentifierSyntax") {
```

Comment:
The first 2 of these contexts were stolen from `TypeSyntax+NameExtractorSpec.swift `, which was delete din this PR. Everything else is pretty new.